### PR TITLE
Generate projects and gists client

### PR DESCRIPTION
**Pull request by hmharvey**
Part of #1157. Generalizes the `openapi_generator` to work with the projects and gists clients. Realized some of the method names I accepted for the previous clients like `issues` aren't specific enough and so there are some retroactive breaking changes, although some are reverting to the original method names. Here's the current breakdown:

### projects.rb
`projects` => `repository_projects`
`create_projects` => `create_repository_projects`
`create_project` => `create_repository_project`
`column_cards` => `project_cards`

new methods for collaborators and helper methods
`project_collaborators`
`add_project_collaborator`
`remove_project_collaborator`
`user_permission_level`
`open_project`
`close_project`

the same
`project`, `projects`, `update_project`, `delete_project`, `org_projects` [but no alias for `organization_projects`], `create_org_projects` [but no alias for `create_organization_projects`], `project_columns`, `create_project_column`,  `project_column`, `update_project_column`, `delete_project_column`, `create_project_card`, `project_card`, `update_project_card`, `move_project_card`, `delete_project_card`

### gists.rb
`gists` => split into `user_public_gists` and `gists`, no alias for `list_gists`
`gist` => split into `gist` and `gist_revision`
`create_gist` has files as a specific parameter
`edit_gist` => `update_gist`
`gist_starred?` => `is_starred?`

the same
`fork_gist`, `gist_forks`,  `gist_commits`, `gist_comment`, `star_gist`, `unstar_gist`, `starred_gists`,  `delete_gists`, `create_gist_comment`, `update_gist_comment`, `delete_gist_comment`, `gist_comments`, `public_gists`

### retroactive changes
`create_comment` => `create_issue_comment`
`update_comment` => `update_issue_comment`
`delete_comment` => `delete_issue_comment`
`add_assignee` => `add_issue_assignee`
`remove_assignee` => `remove_issue_assignee`
`comment` => `issue_comment`
`comments` => `issue_comments`
`add_labels` => `add_issue_labels`
`remove_label` => `remove_issue_label`
`remove_labels` => `remove_issue_labels`
`replace_labels` => `replace_issue_labels`

`label` => `issue_label`
`create_label` => `create_issue_label`
`update_label` => `update_issue_label`
`delete_label` => `delete_issue_label`

`milestone` => `issue_milestone`
`create_milestone` => `create_issue_milestone`
`update_milestone` => `update_issue_milestone`
`delete_milestone` => `delete_issue_milestone`

Also, sorry for rerecording all of the cassettes and having 153 files changed 😬 

TODO:
- [x] clean up methods in generator
- [x] fix preview header generation: read in from previews.rb
- [x] add new project collaborator tests
- [x] double check the tests clean up after themselves

### CODE REVIEW

File: lib/octokit/preview.rb [hunk 1]
```diff
@@ -42,5 +42,15 @@ def warn_preview(type)
 header.
 EOS
     end
+
+    def self.get_media_type(preview)
+      PREVIEW_TYPES.each do |type|
+        if type.last.include? preview
+          return type.first
+        else
+          # octokit needs a new preview
```
Comment (hmharvey) [inline lib/octokit/preview.rb]: I'd love some suggestions for this and what you think of this approach in general 👀 it takes in the preview name used in the Open API spec and returns our Octokit abbreviation.

File: lib/octokit/client/gists.rb [hunk 2]
```diff
@@ -1,233 +1,202 @@
...
       #
-      # @param gist [String] Gist ID
-      # @return [Boolean] Indicating success of deletion
-      # @see https://developer.github.com/v3/gists/#delete-a-gist
-      def delete_gist(gist, options = {})
-        boolean_from_response :delete, "gists/#{Gist.new(gist)}", options
+      # @option options [String] :since This is a timestamp in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format: `YYYY-MM-DDTHH:MM:SSZ`. Only gists updated at or after this time are returned.
+      # @return [Array<Sawyer::Resource>] A list of publics
+      # @see https://developer.github.com/v3/gists/#list-all-public-gists
+      def gist_public(options = {})
```
Comment (tarebyte) [inline lib/octokit/client/gists.rb]: Could we rename this to something like `public_gists` ?

File: lib/octokit/client/hooks.rb [hunk 3]
```diff
@@ -86,28 +86,28 @@ def ping_hook(repo, hook_id, options = {})
         boolean_from_response :post, "#{Repository.path repo}/hooks/#{hook_id}/pings", options
       end
 
-      # Get single org hook
+      # Get single hook
```
Comment (tarebyte) [inline lib/octokit/client/hooks.rb]: Could we add back in the organization descriptor here?

File: lib/octokit/client/issues.rb [hunk 4]
```diff
@@ -82,9 +82,9 @@ def issue(repo, issue_number, options = {})
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
       # @param comment_id [Integer] The ID of the comment
-      # @return [Sawyer::Resource] A list of comment
+      # @return [Sawyer::Resource] A list of comments
       # @see https://developer.github.com/v3/issues/comments/#get-a-single-comment
-      def comment(repo, comment_id, options = {})
+      def issue_comment(repo, comment_id, options = {})
```
Comment (tarebyte) [inline lib/octokit/client/issues.rb]: Looks like this is a bug in the docs, we've opened a PR to fix this.

File: lib/octokit/client/projects.rb [hunk 5]
```diff
@@ -1,314 +1,309 @@
...
+     # @option options [Boolean] :private Sets the visibility of a project board. Setting `private` is only available for organization and user projects. **Note:** Updating a project's visibility requires `admin` access to the project.    Can be one of:  false, true
+     def open_project(project_id, options = {})
```
Comment (tarebyte) [inline lib/octokit/client/projects.rb]: Could we rename this to be `reopen_project`?

File: lib/octokit/client/releases.rb [hunk 6]
```diff
@@ -121,9 +121,9 @@ def delete_release_asset(repo, asset_id, options = {})
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
       # @param tag [String] The tag of the release
-      # @return [Sawyer::Resource] A single release by tag
+      # @return [Sawyer::Resource] A single release
       # @see https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name
-      def release_by_tag(repo, tag, options = {})
+      def tag_release(repo, tag, options = {})
```
Comment (tarebyte) [inline lib/octokit/client/releases.rb]: This name is not super clear, could we revert this? Or find an alternative? This makes it feel like we're tagging a release.

**Pull request merged**